flink/flink.sh

```diff
@@ -133,11 +133,30 @@ EOF
 
 }
 
+# TODO: delete once flink 1.5 is available via apt
+function install_flink_snapshot() {
+  local FLINK_TOPLEVEL='flink-1.5.0'
+  local FLINK_URL='http://www-eu.apache.org/dist/flink/flink-1.5.2/flink-1.5.2-bin-hadoop28-scala_2.11.tgz'
+  local FLINK_LOCAL='/tmp/flink-1.5.0-bin-hadoop28-scala_2.11.tgz'
+  local work_dir="$(mktemp -d)"
+
+  pushd "${work_dir}"
+
+  curl -o "${FLINK_LOCAL}" "${FLINK_URL}"
+  tar -xzvf "${FLINK_LOCAL}"
+  mv "${FLINK_TOPLEVEL}" "${FLINK_INSTALL_DIR}"
+
+  popd # work_dir
+
+}
+
 function main() {
 local role="$(/usr/share/google/get_metadata_value attributes/dataproc-role)"
 if [[ "${role}" == 'Master' ]] ; then
   update_apt_get || err "Unable to update apt-get"
-  apt-get install -y flink || err "Unable to install flink"
+  # TODO: re-enable once flink 1.5 is available via apt
+  # apt-get install -y flink || err "Unable to install flink"
+  install_flink_snapshot || err "Unable to install flink"
```

Comment:
I don't think we should install this by default for all users of Flink. Please make this configurable, e.g., by GCE metadata.

In addition, please add documentation of the new configuration option on the Flink README _and_ in the Beam README/init action (pointing out whatever version is required).

Reply:
These are all good points.  Fixed.